modules/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -196,13 +170,13 @@ public HttpRequestInitializer getHttpRequestInitializer(ServiceOptions<?, ?> ser
 
                 return (httpRequest) -> {
                     if (requestInitializer != null) requestInitializer.initialize(httpRequest);
-                    httpRequest.setResponseInterceptor(httpStatsCollector);
+                    httpRequest.setResponseInterceptor(new MeteringResponseInterceptor());
```

Comment:
Nit: `MeteringResponseInterceptor` looks stateless, could we just have a singleton instance?